src/cfnlint/__init__.py

```diff
@@ -150,16 +151,18 @@ def matchall_resource_sub_properties(self, filename, cfn, resource_properties, p
 class RulesCollection(object):
     """Collection of rules"""
 
-    def __init__(self, ignore_rules=None, include_rules=None, configure_rules=None, include_experimental=False):
+    def __init__(self, ignore_rules=None, include_rules=None, configure_rules=None, include_experimental=False, mandatory_rules=None):
         self.rules = []
 
         # Whether "experimental" rules should be added
         self.include_experimental = include_experimental
 
         # Make Ignore Rules not required
         self.ignore_rules = ignore_rules or []
+        self.mandatory_rules = mandatory_rules or []
         self.include_rules = include_rules or []
         self.configure_rules = configure_rules or []
+        self.rules_trie = StringTrie()
```

Comment:
@kddejong What are your thoughts on using a data structure such as trie in order to leverage for comparisons? This, I think, would be a good kick off point for doing prefix unwrapping and make it simpler to handle overlap between prefixed exclusions/mandatory_checks.